### PR TITLE
Fixes #14160: Rewrite and simplify variableFromJsonFile to support separated policy generation

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/14160

### CODE REVIEW

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
Temp file removal is missing!

File: techniques/systemSettings/misc/variableFromJsonFile/3.0/variableFromJsonFile.st [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+bundle agent variable_from_json_file_RudderUniqueID
+{
+  vars:
+    &VARIABLE_FROM_JSON_FILE_JSON_FILE:{json_file |  "json_file[&i&]"        string => "&json_file&";
+}&
+    &TRACKINGKEY:{piuuid |  "trackingkey[&i&]"      string => "&piuuid&";
+}&
+    &VARIABLE_FROM_JSON_FILE_SOURCE:{source |  "sources[&i&]"      string => "&source&";
+}&
+    "complete_name" string => "&VARIABLE_FROM_JSON_FILE_VARIABLE_NAME&";
```
Comment (ncharles) [inline techniques/systemSettings/misc/variableFromJsonFile/3.0/variableFromJsonFile.st]: indentation is 2 spaces for promiser type, 4 for classes condition, and 6 for promiser

File: techniques/systemSettings/misc/variableFromJsonFile/3.0/variableFromJsonFile.st [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    pass3::
+      "remote_base_file_ok" expression => "base_remote_source_list_${files_index}.file_copy_from_remote_source_${base_source_file_canon}_${base_destination_file_canon}_ok.variable_dict_from_file_${complete_name_canon}_${base_destination_file_canon}_ok";
+      "local_base_file_ok"  expression => "base_local_source_list_${files_index}.file_copy_from_local_source_${base_file_canon}_${base_destination_file_canon}_ok.variable_dict_from_file_${complete_name_canon}_${base_destination_file_canon}_ok";
+      "base_file_ok"        expression => "remote_base_file_ok|local_base_file_ok";
+
+
+  methods:
+    pass1::
+    #Load the differents Json File
+    "prepare" usebundle => directory_create("${directory_agent}");
```
Comment (ncharles) [inline techniques/systemSettings/misc/variableFromJsonFile/3.0/variableFromJsonFile.st]: could you fix indentation here also ?

**Comment by Fdall** [comment]
Commit modified

File: techniques/systemSettings/misc/variableFromJsonFile/3.0/variableFromJsonFile.st [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      "json_file_canon[${files_index}]"      string => canonify("${json_file[${files_index}]}");
+
+      "name_list[${files_index}]"            string => "${complete_name}";
+      "base_source_list[${files_index}]"     string => "${base_source}";
+      "base_file_list[${files_index}]"       string => "${base_file}";
+      "base_file_canon_list[${files_index}]" string => "${base_file_canon}";
+
+      # Split the variable complete name in prefix and variable name
+      "strTok" slist => splitstring("${complete_name}", "\.", "2");
+      "variable_prefix" string => nth("strTok", 0);
```
Comment (ncharles) [inline techniques/systemSettings/misc/variableFromJsonFile/3.0/variableFromJsonFile.st]: to improve readability, could you also aling the => ?

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
OK, merging this PR

**Pull request merged**